NVIDIA/NV-Kernels
# feat: [DGX-8847] Backport MPAM patch to 24.04-linux-nvidia-adv-6.8-next

BugLink: https://jirasw.nvidia.com/browse/DGX-8847  

[Description]
Backport MPAM patches from MPAM-dev[0] 
Also tried to backport MPAM patches based on PR25 to branch [1]

[0], https://github.com/NVIDIA-BaseOS-6/linux-nvidia-6.5/tree/MPAM-dev
[1], https://github.com/KobaKoNvidia/NV-Kernels/tree/kobak_dgx8849_pr-nv-kernels-25_backportMPAM

[Test Procedure]  
1. mount mpam resctrl and create partition P2  
2. Run stress-ng  and assign PIDs to p2/task
```
$ killall stress-ng ; stress-ng --stream 144 --stream-l3-size 114m --stream-madvise hugepage & pidof stress-ng | xargs -rn1 echo | sudo tee /sys/fs/resctrl/p2/task
```
3. Use perf to monitor MB
```
$ sudo ./perf stat -I 1000 -e resctrl_pmu/mbm_local_bytes,resctrl_id=`cat /sys/fs/resctrl/p2/id`,domain=1/
```
4.  Check Cache Portion 
```
$ cat /sys/fs/resctrl/p2/mon_data/mon_L3_0*/llc_occupancy
```
5. Check [1] for the details
[2]. [Test Results after backporting patches to linux-nvidia-adv-6.8-next](https://confluence.nvidia.com/display/DPS/Test+Results+after+backporting+patches+to+linux-nvidia-adv-6.8-next)
[3] Test Results with Lego-CG1-QS-38, https://confluence.nvidia.com/display/DPS/Test+Results+after+backporting+patches+to+linux-nvidia-adv-6.8-next

Jira: DGX-8847
ChangeLog: feat
